src/Swift.Bindings/src/Model/TypeNames/SwiftTypeName.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Represents a Swift type name.
+/// </summary>
+public record SwiftTypeName
+{
+    /// <summary>
+    /// The module name.
+    /// </summary>
+    public string Module { get; }
+
+    /// <summary>
+    /// The type name.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// The module-qualified type name. Includes all parent types.
+    /// </summary>
+    public string ModuleQualifiedName { get; }
+
+    /// <inheritdoc />
+    public override string ToString() => ModuleQualifiedName;
+
+    private SwiftTypeName(string module, string name, string moduleQualifiedName)
+    {
+        Module = module;
+        Name = name;
+        ModuleQualifiedName = moduleQualifiedName;
+    }
+
+    /// <summary>
+    /// Creates a new SwiftTypeName from a module-qualified name.
+    /// </summary>
+    /// <param name="moduleQualifiedName">The module-qualified name.</param>
+    /// <returns>The SwiftTypeName.</returns>
+    public static SwiftTypeName FromModuleQualifiedName(string moduleQualifiedName)
```

Comment:
Does this function need to handle generics?

Reply:
Once we have a mature generic types solution - no. The type database would contain open generic types. And when a new closed generic is encountered a special handler would dissect into a set of nominal types (open generic, type for each generic paramater etc). Then each of the nominal types could be read from the database. 

Currently we keep some generic types in the type database (as a workaround) so this will be invoked with generic params (they are not parsed though) so e.g. "Module.Foo<UInt8> "would just be hashed and used to query the type database.